etsd/keys/views.py

```diff
@@ -44,6 +47,34 @@ def get_context_data(self, *args, **kwargs):
 class PublicKeyDetailView(AdminOrAuthorityQsMixin, DetailView):
     model = models.PublicKey
 
+    def post(self, request, *args, **kwargs):
+        pubk= models.PublicKey.objects.get(pk=self.kwargs.get('pk'))
+
+        if "acceptKey" in request.POST:
+            activekeys = models.PublicKey.objects.filter(authority=pubk.authority, status="ACTIVE")
+            for key in activekeys:
+                key.status = "INACTIVE"
+                key.save()
+            pubk.approved_on = datetime.now()
```

Comment:
Never use `datetime.now()` because it returns a naive datetime (no timezone information). Use djang.utils.timezone.now() instead.